plugins/callback/logstash.py

```diff
@@ -63,76 +92,85 @@
+                self.ls_server,
+                self.ls_port,
+                version=1,
+                message_type=self.ls_type
+            )
+
+            self.logger.addHandler(self.handler)
+            self.hostname = socket.gethostname()
+            self.session = str(uuid.uuid4())
+            self.errors = 0
+
+            self.base_data = {
+                'session': self.session,
+                'host': self.hostname
+            }
+
+            if self.ls_pre_command is not None:
+                self.base_data['ansible_pre_command_output'] = os.popen(
+                    self.ls_pre_command).read()
 
+            if self._options is not None:
+                self.base_data['ansible_checkmode'] = self._options.check
+                self.base_data['ansible_tags'] = self._options.tags
+                self.base_data['ansible_skip_tags'] = self._options.skip_tags
+                self.base_data['inventory'] = self._options.inventory
+
+    def set_options(self, task_keys=None, var_options=None, direct=None):
         super(CallbackModule, self).set_options(task_keys=task_keys, var_options=var_options, direct=direct)
 
-        self.logger = logging.getLogger('python-logstash-logger')
-        self.logger.setLevel(logging.DEBUG)
-
-        self.logstash_server = self.get_option('server')
-        self.logstash_port = self.get_option('port')
-        self.logstash_type = self.get_option('type')
-        self.handler = logstash.TCPLogstashHandler(
-            self.logstash_server,
-            int(self.logstash_port),
-            version=1,
-            message_type=self.logstash_type
-        )
-        self.logger.addHandler(self.handler)
-        self.hostname = socket.gethostname()
-        self.session = str(uuid.uuid1())
-        self.errors = 0
+        try:
+            self.ls_server = self.get_option('server')
+        except KeyError:
+            self.ls_server = "localhost"
+
+        try:
+            self.ls_port = int(self.get_option('port'))
+        except KeyError:
+            self.ls_port = 5000
+
+        try:
+            self.ls_type = self.get_option('type')
+        except KeyError:
+            self.ls_type = "ansible"
+
+        try:
+            self.ls_pre_command = self.get_option('pre_command')
+        except KeyError:
+            pass
+
+        self._init_plugin()
 
     def v2_playbook_on_start(self, playbook):
-        self.playbook = playbook._file_name
-        data = {
-            'status': "OK",
-            'host': self.hostname,
-            'session': self.session,
-            'ansible_type': "start",
-            'ansible_playbook': self.playbook,
-        }
-        self.logger.info("ansible start", extra=data)
+        data = self.base_data.copy()
+        data['ansible_type'] = "start"
+        data['status'] = "OK"
+        data['ansible_playbook'] = playbook._file_name
+        self.logger.info("START PLAYBOOK | %s", data['ansible_playbook'], extra=data)
```

Comment:
You are changing the text that is logged. Please remove that from this PR, and add it to another PR (that will be a breaking change).

(Same for the similar lines.)

Reply:
I plan to add property `format_version` for switching new format (like "START PLAYBOOK | bla bla") and legacy

Reply:
format can be v1 or v2 (default will be v1)

Reply:
That works for me!